pkg/apis/vpc/v1alpha1/subnet_types.go

```diff
@@ -25,6 +25,9 @@ const (
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.accessMode) || has(self.accessMode)", message="accessMode is required once set"
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.ipAddresses) || has(self.ipAddresses)", message="ipAddresses is required once set"
 type SubnetSpec struct {
+	// VPC name of the Subnet
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	VPCName string `json:"vpcName,omitempty"`
```

Comment:
will it be set for existing subnets in the primary VPC?

Reply:
Yes，we will also set vpcName for existing Subnets when NCP is upgraded.

Reply:
Then would the validation `rule="self == oldSelf"` reject the update?

Reply:
good point。Then I think for "vpcName" we can leave it empty for existing Subnet CR. If empty, that means the Subnet CR is in Namespace default VPC. Do you think this sounds good?

Reply:
Is there an expression to make it immutable once set?

Reply:
// +kubebuilder:validation:XValidation:rule="!has(oldSelf) || self == oldSelf",message="Value is immutable after set"

changed to above. We will have a webhook to only allow NSX Operator can update "vpcName" field.